WordPress/openverse
# Remove MediaViewSet's lookup regex

## Description


Since the upstream issue (encode/django-rest-framework#6789) is fixed, it's safe to remove this workaround to handle invalid UUID-like strings. This way, DRF handles better these requests and responses are consistent with the rest.

### Response in production

https://api.openverse.org/v1/images/000000000000000000000000000000000000/related/

![Screenshot 2024-12-12 at 12-35-51 Not Found](https://github.com/user-attachments/assets/5de71cf7-cd14-4be8-81fc-e1f7909ced40)

### Response in this branch

http://localhost:50280/v1/images/000000000000000000000000000000000000/related/

![Screenshot 2024-12-12 at 12-32-37 Related – Django REST framework](https://github.com/user-attachments/assets/9cdef1a9-11e1-4ddb-9235-a54a52eb1c4a)

## Testing Instructions


Run the API, `just api/up`, and check other invalid UUIDs from the original issue (WordPress/openverse-api#729) if you wish.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
